modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesKernelBoot.java

```diff
@@ -335,10 +343,35 @@ public static void startHouseKeepingThreads() throws AndesException {
 
         //reload exchanges/queues/bindings and subscriptions
         AndesRecoveryTask andesRecoveryTask = new AndesRecoveryTask();
-        Integer scheduledPeriod = AndesConfigurationManager.readValue
+        //deleted the expired message from db
+        PeriodicExpiryMessageDeletionTask periodicExpiryMessageDeletionTask = null;
+
+        int recoveryTaskScheduledPeriod = AndesConfigurationManager.readValue
                 (AndesConfiguration.PERFORMANCE_TUNING_FAILOVER_VHOST_SYNC_TASK_INTERVAL);
-        andesRecoveryTaskScheduler.scheduleAtFixedRate(andesRecoveryTask, scheduledPeriod,
-                                                       scheduledPeriod, TimeUnit.SECONDS);
+        int dbBasedDeletionTaskScheduledPeriod = AndesConfigurationManager.readValue
+                (AndesConfiguration.PERFORMANCE_TUNING_PERIODIC_EXPIRY_MESSAGE_DELETION_INTERVAL);
+        int safeDeleteRegionSlotCount = AndesConfigurationManager.readValue
+                (AndesConfiguration.PERFORMANCE_TUNING_SAFE_DELETE_REGION_SLOT_COUNT);
+        boolean isDLCExpiryCheckEnabled = AndesConfigurationManager.readValue
+                (AndesConfiguration.PERFORMANCE_TUNING_EXPIRE_MESSAGES_IN_DLC);
+
+        //based on the DLC expiration check configuration bind the appropriate deletion task
+        if (isDLCExpiryCheckEnabled) {
+            periodicExpiryMessageDeletionTask = new DLCExpiryCheckEnabledDeletionTask();
+        } else {
+            periodicExpiryMessageDeletionTask = new PeriodicExpiryMessageDeletionTask();
+        }
+
+        andesRecoveryTaskScheduler.scheduleAtFixedRate(andesRecoveryTask, recoveryTaskScheduledPeriod,
+                                                       recoveryTaskScheduledPeriod, TimeUnit.SECONDS);
+        if (safeDeleteRegionSlotCount >= 1) {
+            expiryMessageDeletionTaskScheduler.scheduleAtFixedRate(periodicExpiryMessageDeletionTask,
+                    dbBasedDeletionTaskScheduledPeriod, dbBasedDeletionTaskScheduledPeriod, TimeUnit.SECONDS);
+        } else {
+            log.warn("DB based expiry message deletion task is not scheduled due to not providing " +
```

Comment:
The idea was that we issue a warning and continue starting the server without initializing the expiration task. Should the log just be an 'error' log, or do we also need to stop the server from starting up?